File: frontend/src/services/CheckInService.js
import axios from 'axios';

const API_URL = 'http://localhost:8080/api/check_in';

class CheckInService {
    getAllCheckins() {
        return axios.get(API_URL);
    }

    getCheckinById(id) {
        return axios.get(`${API_URL}/${id}`);
    }

    createCheckin(checkin) {
        return axios.post(API_URL, checkin);
    }

    updateCheckin(id, checkin) {
        return axios.put(`${API_URL}/${id}`, checkin);
    }

    deleteCheckin(id) {
        return axios.delete(`${API_URL}/${id}`);
    }
}

export default new CheckInService();

File: src/main/java/Airlines/AirlineTicket/CheckIn/Models/CheckInService.java
package Airlines.AirlineTicket.CheckIn.Models;
import Airlines.AirlineTicket.CheckIn.CheckIn;
import java.util.List;

public interface CheckInService {

    public List<CheckIn> getAllCheckins();

    public CheckIn getCheckinById(Long id);

    public CheckIn saveCheckin(CheckIn checkin);

    public void deleteCheckin(Long id);
}